src/mechanics/tools/QuasistaticProblem.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "mechanics/tools/EquationSystem.h"
+#include "mechanics/constraints/Constraints.h"
+
+namespace NuTo
+{
+class QuasistaticProblem
+{
+public:
+    //! Ctor
+    //! @param equations system of equations including Gradient(), Hessian0() and UpdateHistory()
+    //! @param dof dof type
+    QuasistaticProblem(EquationSystem& equations, DofType dof);
```

Comment:
In this special situation, where we essentially do not need any time integration scheme, this works well. However, for generalization purposes, I think QuasiStaticProblem should also provide a Hessian and Gradient method that is then called in the DoStep method (where itself the newton iteration is used). However, this can certainly also done later when trying to generalize this. Futhermore, I don't fully understand the need for a distinction between EquationSystem (apparently used to call the Newton iteration) and the QuasiStaticProblem (which additionally includes the constraints). But this is probably due to me thinking of throwing out the constraints from the problem definition.

Reply:
You commented on an outdated file. But I agree with your point of having Hessian and Gradient as methods. They currently do exist and are called `Residual` and `Derivative`. The problem here is that the newton algorithm does not care about time. So the `Residual` function is called by the newton raphson algorithm. This method calls `Gradient(..., t)` from the TimeDependentProblem and the time `t` has to be provided somehow. This makes everything a bit messy.